clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -56,7 +57,9 @@ public SaslChannelBuilder(Mode mode, LoginType loginType, SecurityProtocol secur
     public void configure(Map<String, ?> configs) throws KafkaException {
         try {
             this.configs = configs;
-            this.loginManager = LoginManager.acquireLoginManager(loginType, configs);
+            String mechanismName = (String) this.configs.get(SaslConfigs.SASL_MECHANISM);
+            mechanism = mechanismName == null ? SaslMechanism.GSSAPI : SaslMechanism.fromMechanismName(mechanismName);
```

Comment:
We are setting the default in the config already, we should not duplicate it here IMO. If the config is missing, we can probably throw an IllegalArgumentException.